frontend/packages/dev-console/src/utils/pipeline-actions.tsx

```diff
@@ -175,10 +175,17 @@ export const stopPipelineRun: KebabAction = (kind: K8sKind, pipelineRun: Pipelin
   return {
     label: 'Stop',
     callback: () => {
-      k8sUpdate(PipelineRunModel, {
-        ...pipelineRun,
-        spec: { ...pipelineRun.spec, status: 'PipelineRunCancelled' },
-      });
+      // Fetching the latest resourceVersion for the Pipelinerun to avoid 409 conflict
+      k8sGet(PipelineRunModel, pipelineRun.metadata.name, pipelineRun.metadata.namespace)
+        .then((latestPLR) =>
+          k8sUpdate(PipelineRunModel, {
```

Comment:
Why is this not using `patch`?

We don't want to get into a pattern of fetching resources before updating. It doesn't actually solve any problem. Only further mitigates an already low percent chance of error. Actually we already do this when firehose is being used.

Reply:
@christianvogt Updated to use patch. Works well after removing the resourceVersion from the equation.